.circleci/config.yml

```diff
@@ -755,9 +800,10 @@ jobs:
     <<: *steps-checkout
 
   mac-checkout:
-    <<: *machine-mac-large
+    <<: *machine-linux-2xlarge
     environment:
-      <<: *env-mac-large
+      <<: *env-linux-2xlarge
+      GCLIENT_EXTRA_ARGS: '--custom-var=checkout_mac=True --custom-var=host_os=mac'
```

Comment:
I think it should be `target_os` and `target_os_only` instead https://cs.chromium.org/chromium/tools/depot_tools/gclient.py?l=64-81. `checkout_mac`will get the right value based on it.

Reply:
If this doesn't fix getting the right binaries as a part of gclient hooks, then I am nervous about this change. I would like to avoid re-downloading the right binaries in a separate step, its not worth the benefit of a checkout time saving.

Reply:
I tried `target_os` and it doesn't do what we need.  We need it think it's running on macOS not building for macOS.

I would argue that 15+ minutes less build time is 100% worth re-downloading the clang binaries

Reply:
Its not just clang binary, are other binaries and buildtools chromium downloads during the hooks correspond to the right os ? If thats unaffected then I am good with this change.

Reply:
Cool, thanks for confirming!

Reply:
wrong clang binary when setting `host_os` explicitly seems like a bug upstream. can we fix it there?

Reply:
@nornagon It's something we can follow up on, I looked into supporting it but the file that downloads clang uses `sys.platform` *everywhere* 😄 Patchable but not required to make this work 👍